app/containers/MessageActions.js

```diff
@@ -87,7 +89,7 @@ class MessageActions extends React.Component {
 		}
 
 		// Pin
-		if (Message_AllowPinning) {
+		if (Message_AllowPinning && !isDirectMessage) {
```

Comment:
That's not quite right.
Pin is should work on direct messages.

We should check for pin-message permission instead of the room type.
https://github.com/RocketChat/Rocket.Chat/blob/develop/app/message-pin/client/actionButton.js#L57